contracts/allowlist/OperatorAllowlistUpgradeable.sol

```diff
@@ -0,0 +1,167 @@
+// Copyright Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: Apache 2.0
+pragma solidity 0.8.19;
+
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+// Introspection
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+// Interfaces
+import {IOperatorAllowlist} from "./IOperatorAllowlist.sol";
+import {IWalletProxy} from "./IWalletProxy.sol";
+
+/*
+    OperatorAllowlist is an implementation of a Allowlist registry, storing addresses and bytecode
+    which are allowed to be approved operators and execute transfers of interfacing token contracts (e.g. ERC721/ERC1155).
+    The registry will be a deployed contract that tokens may interface with and point to.
+*/
+
+contract OperatorAllowlistUpgradeable is ERC165, AccessControlUpgradeable, UUPSUpgradeable, IOperatorAllowlist {
```

Comment:
We must used AccessControlEnumerableUpgradeable
Being able to enumerate who the admins are is very important.